src/chunkserver/chunkserver-common/default_disk_manager.cc

```diff
@@ -304,3 +304,31 @@ IDisk *DefaultDiskManager::getDiskForNewChunk(
 
 	return bestDisk;
 }
+
+IDisk *DefaultDiskManager::getDiskForGC() {
+	TRACETHIS();
+	IDisk *bestDisk = DiskNotFound;
+
+	std::lock_guard disksLockGuard(gDisksMutex);
+
+	if (gDisks.empty()) {
+		return DiskNotFound;
+	}
+
+	auto diskCount = gDisks.size();
+
+	for (size_t i = 0; i < diskCount; ++i) {
+		size_t index = (lastDiskIndexForGC + i) % diskCount;
+		const auto &disk = gDisks[index];
+
+		if (!disk->isZonedDevice() || !disk->isSelectableForNewChunk()) {
+			continue;
+		}
+
+		bestDisk = disk.get();
+		lastDiskIndexForGC = (index + 1) % diskCount;
```

Comment:
The update of `lastDiskIndexForGC` equals to `index + 1` suggests to me that the stored value is actually the `nextDiskIndexForGC`, i.e., the initial index considered in the next iteration this function will be called, because the returned disk is the one at `index` position.

Reply:
Nice catch, thanks. Renamed to nextDiskIndexForGC_.